Makefile

```diff
@@ -12,34 +12,26 @@ endif
 LDFLAGS +=  -lz
 
 all: submodules release debug
-debug: target/debug/librubyfmt.a target/debug/rubyfmt-main
-release: target/release/librubyfmt.a target/release/rubyfmt-main
 
-target/debug/rubyfmt-main: librubyfmt/src/*.rs librubyfmt/Cargo.toml src/*.rs Cargo.toml
+debug:
 	cargo build
 
-target/release/rubyfmt-main: librubyfmt/src/*.rs librubyfmt/Cargo.toml src/*.rs Cargo.toml
+release:
 	cargo build --release
 
 submodules:
 	git submodule init
 	git submodule update
 
-target/c_main_debug: main.c target/debug/librubyfmt.a
-	clang -O3 main.c target/debug/librubyfmt.a $(LDFLAGS) -o $@
+target/c_main_debug: target/debug/deps/librubyfmt-*.a main.c
+	clang -O3 main.c $< $(LDFLAGS) -o $@
 
-target/c_main_release: main.c target/release/librubyfmt.a
-	clang -O3 main.c target/release/librubyfmt.a $(LDFLAGS) -o $@
+target/c_main_release: target/release/deps/librubyfmt-*.a main.c
+	clang -O3 main.c $< $(LDFLAGS) -o $@
 
-target/release/librubyfmt.a: librubyfmt/src/*.rs librubyfmt/Cargo.toml
-	mkdir -p target/release
-	cd librubyfmt && cargo build --release
-	cp librubyfmt/target/release/librubyfmt.a $@
+target/release/deps/librubyfmt-*.a: release
 
-target/debug/librubyfmt.a: librubyfmt/src/*.rs librubyfmt/Cargo.toml
-	mkdir -p target/debug
-	cd librubyfmt && cargo build
-	cp librubyfmt/target/debug/librubyfmt.a $@
+target/debug/deps/librubyfmt-*.a: debug
```

Comment:
I was concerned that using a wildcard for these would be a problem, but I did a few cycles of making a small change and rebuilding and there was only ever one file matching this pattern at once.